src/core/options-registry.cpp

```diff
@@ -0,0 +1,122 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include "options-registry.h"
+#include "basics.h"
+#include <src/librealsense-exception.h>
+#include <vector>
+#include <mutex>
+#include <map>
+
+
+namespace librealsense {
+
+
+LRS_EXTENSION_API std::string const & get_string( rs2_option );
+
+
+namespace by_name {
+
+
+static std::map< std::string, rs2_option > _option_by_name = []()
+{
+    // Automatically add all our known options into the option names -- that way, no duplicate custom option
+    // can be added...
+    std::map< std::string, rs2_option > map;
+    for( auto option = rs2_option( 0 ); option < RS2_OPTION_COUNT; option = rs2_option( option + 1 ) )
+        map[get_string( option )] = option;
+    return std::move( map );
+}();
+static std::vector< std::string > _name_by_index;
+static std::mutex _mutex;
+
+
+inline size_t option_to_index( rs2_option by_name_option )
+{
+    return -by_name_option - 1;
+}
+
+
+inline rs2_option index_to_option( size_t index )
+{
+    return rs2_option( -( int( index ) + 1 ) );
```

Comment:
I see now you limit the number of options. Use a constant for that limit and check here too.

Reply:
It's called right after `new_index`, which is the only place right now that has the concept of a limit.
This is just a conversion function, I don't think it's the place to add additional logic.